core/iwasm/common/wasm_memory.c

```diff
@@ -975,7 +969,8 @@ wasm_enlarge_memory(WASMModuleInstance *module, uint32 inc_page_count)
     if (module->memory_count > 0)
         shared_memory_lock(module->memories[0]);
 #endif
-    ret = wasm_enlarge_memory_internal(module, inc_page_count, 0);
+    ret = wasm_enlarge_memory_internal((WASMModuleInstanceCommon *)module,
+                                       module->memories[0], inc_page_count);
```

Comment:
Since it's an export API, we also need to check whether `module->memories` is valid.